game-core/src/main/java/games/strategy/engine/data/ProductionRuleList.java

```diff
@@ -6,6 +6,9 @@
 
 import com.google.common.annotations.VisibleForTesting;
 
+/**
+ * A keyed collection of {@link ProductionRule}s.
```

Comment:
This javadoc is quite useful.
IMO could go further, be more complete -  what is the collection keyed by?